cellxgene_resources/curation_sample_code.ipynb

```diff
@@ -651,6 +790,58 @@
     "**IN PROGRESS: If a reference other than one of the default CellRanger reference files was used, a map file can be created from the annotation file for use in curation**"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**If a GENCODE/ENSEMBL reference was used, parse the annotation file**"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Use pd.read_table to parse the gtf & apply various manipulations to produce symbol-ENSG table\n",
+    "gtf = pd.read_table('path/to/gtf', header=None)\n",
+    "gene_dict = {\n",
+    "        'gene_ids': [],\n",
+    "        'gene_symbols': []\n",
+    "    }\n",
+    "\n",
+    "# extract gene_ids and gene_symbols from last column in the gtf table\n",
+    "for index,row in gtf.iterrows():  \n",
+    "    ensg_id = row[8].split(';')[0].split('\"')[1].split('.')[0]\n",
+    "    if ensg_id not in gene_dict['gene_ids']:\n",
+    "        print(ensg_id)\n",
+    "        gene_dict['gene_ids'].append(ensg_id)\n",
+    "        symbol = row[8].split(';')[3].split('\"')[1]\n",
+    "        print(symbol)\n",
+    "        gene_dict['gene_symbols'].append(symbol)\n",
+    "\n",
+    "# if gene_ids and gene_symbols lists are of different lengths, it's likely that there are duplicated/multiple gene_symbols, also check for empty strings or null values \n",
+    "# # Dealing with duplicates: assign 'multiple' to duplicated gene_symbols\n",
+    "dups = annotations[annotations.duplicated(subset='gene_symbols',keep=False)]\n",
+    "dups['gene_ids'] = 'multiple'\n",
+    "dups = dups.drop_duplicates()\n",
+    "print(f\"There were {dups.shape[0]} duplicated gene_symbols\")\n",
+    "annotations = annotations.drop_duplicates(subset='gene_symbols',keep=False)\n",
+    "annotations = annotations.append(dups)\n",
+    "annotations.set_index('gene_ids', inplace=True, )\n",
+    "\n",
+    "# create dataframe from dictionary\n",
+    "annotations = pd.DataFrame(gene_dict)  \n",
+    "print(annotations.shape)\n",
+    "print(annotations['gene_symbols'].value_counts(dropna=False))\n",
+    "\n",
+    "# write dataframe to tsv file in ref_file folder\n",
+    "path = 'path/to/lattice_curation/ref_files/'\n",
+    "file = 'refdata-[source]-[ref_version]_genes_gtf.tsv'  # change reference version and source as needed\n",
```

Comment:
The refdata-* format was simply derived from an input gtf that started refdata-*
So in order to prevent the user from needing to update another variable and to maintain provenance from the gtf that this table was created from, I would just manipulate the name of the input gtf to get what you want...
`file = gtf_file.replace('.','_') + '.tsv'`
so that *.gtf just becomes *_gtf.tsv